Gems/Multiplayer/Code/Include/Multiplayer/IMultiplayerDebug.h

```diff
@@ -7,9 +7,64 @@
  */
 
 #pragma once
+#include <AzCore/Math/MathStringConversions.h>
 
 namespace Multiplayer
 {
+    enum class MultiplayerAuditCategory
+    {
+        Desync,
+        Input,
+        Event
+    };
+
+    //! @class IMultiplayerAuditingDatum
+    //! @brief IMultiplayerAuditingDatum provides an interface for auditing datums of a given auditing event
+    class IMultiplayerAuditingDatum
+    {
+    public:
+        virtual ~IMultiplayerAuditingDatum() = default;
+        virtual IMultiplayerAuditingDatum& operator= (const IMultiplayerAuditingDatum&) { return *this; }
+
+        virtual const AZStd::string& GetName() const = 0;
+        virtual AZStd::pair<AZStd::string, AZStd::string> GetClientServerValues() const = 0;
+        virtual AZStd::unique_ptr<IMultiplayerAuditingDatum> Clone() = 0;
+    };
+
+    //! @class MultiplayerAuditingElement
+    //! @brief MultiplayerAuditingElement contains a list of datums for a given auditing event
+    class MultiplayerAuditingElement
+    {
+    public:
+        MultiplayerAuditingElement() = default;
+        MultiplayerAuditingElement(const MultiplayerAuditingElement& rhs)
+        {
+            name = rhs.name;
+            elements.resize(rhs.elements.size());
+            for (int32_t i = 0; i < rhs.elements.size(); ++i)
+            {
+                elements[i] = rhs.elements[i]->Clone();
+            }
+        }
+
+        MultiplayerAuditingElement& operator= (const MultiplayerAuditingElement& rhs)
+        {
+            if (&rhs != this)
+            {
+                name = rhs.name;
+                elements.resize(rhs.elements.size());
+                for (int32_t i = 0; i < rhs.elements.size(); ++i)
+                {
+                    elements[i] = rhs.elements[i]->Clone();
+                }
+            }
+            return *this;
+        }
+
+        AZStd::string name;
```

Comment:
Why break encapsulation by exposing member variables? These should also be m_ as per C++ coding standards (https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#variable-names)

Reply:
This really should be a struct as it's an abstract value data container. Will fix the naming as well.